hackathon_site/event/views.py

```diff
@@ -200,3 +207,38 @@ def post(self, request, *args, **kwargs):
         at once.
         """
         return super().post(request, *args, **kwargs)
+
+
+class LeaveTeamView(generics.GenericAPIView):
+    permission_classes = [UserHasProfile]
+
+    @transaction.atomic
+    @swagger_auto_schema(responses={201: TeamSerializer})
+    def post(self, request, *args, **kwargs):
+        profile = request.user.profile
+        team = profile.team
+
+        # Raise 400 if team has active orders
+        active_orders = OrderItem.objects.filter(
+            Q(part_returned_health__isnull=True),
```

Comment:
Hmm do we want these two conditions? I would originally have thought if they had an order at all, lock them to the team.

I don't want there to be the case where they somehow leave the team after the event (when everything is returned), screwing up our metrics. Maybe only allow them to leave if no parts have actually been checked out (ie only cancelled orders allowed)? That would just be removing this first clause.